lib/simplecov/source_file.rb

```diff
@@ -72,13 +73,105 @@ def status
       end
     end
 
+    #
+    # Representing single branch that has been detected in coverage report.
+    # Give us support methods that handle needed calculations.
+    class Branch
+      include SimpleCov::Supports::BranchSupport
+
+      attr_reader :type,
+                  :id,
+                  :start_line,
+                  :start_col,
+                  :end_line,
+                  :end_col
+
+      attr_accessor :coverage, :root_id
+
+      def initialize(*args)
+        @type       = args[0]
+        @id         = args[1]
+        @start_line = args[2]
+        @start_col  = args[3]
+        @end_line   = args[4]
+        @end_col    = args[5]
+        @root_id    = args[6]
+        @coverage   = 0
+      end
+
+      #
+      # Return true if there is relevant count defined > 0
+      #
+      # @return [Boolean]
+      #
+      def covered?
+        coverage.positive?
+      end
+
+      #
+      # Check if branche missed or not
+      #
+      # @return [Boolean]
+      #
+      def missed?
+        coverage.zero?
+      end
+
+      #
+      # Current branch is root or not
+      #
+      # @return [Boolean]
+      #
+      def root?
+        root_id.nil?
+      end
+
+      #
+      # Current branch is sub_branch
+      #
+      # @return [Boolean]
+      #
+      def sub_branch?
+        !root?
+      end
+
+      #
+      # Branch is positive or negative.
+      # For `case` conditions, `when` always supposed as positive branch.
+      #
+      # @return [Boolean]
+      #
+      def positive?
+        return true if type == :when
+        1 + root_id.to_i == id
```

Comment:
Highlighting this once more cos I still got no idea what's happening here :grin:

Reply:
For `if, else` conditions:
coverage returns matrices ex: [:if, 0,..] => {[:then, 1,..], [:else, 2,..]},
where positive branch always has Id equals to root branch Id incremented by 1.

Reply:
I will add more comments to this method

Reply:
Guessed as much but wanted confirmation. Thank you! I guess it also works for `unless` which then is the question should that still be called positive but I think so. Thank you!